### PR TITLE
[My Tasks] feat: simple My Tasks flow

### PR DESCRIPTION
**Pull request by jeannineruiz**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
I created a simple flow for My Tasks with placeholder data. You can see what donation tasks you have and enter data for them as well as mark them as complete or missed.

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Main pages are my-tasks/_layout and donation-details/_layout

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

The picker for the NPOs needs to be styled properly. 

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @insert pl github username here

### CODE REVIEW

**Comment by stephenhungg** [comment]
looks good, consider making the alert into a toast instead

File: src/app/donation-details/_layout.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  },
+  input: {
+    borderWidth: 1,
+    borderColor: '#ccc',
+    borderRadius: 12,
+    padding: 12,
+    backgroundColor: 'white',
+    marginBottom: 15,
+  },
+  pickerContainer: {
```
Comment (kevin3656) [inline src/app/donation-details/_layout.tsx]: I dont think this was used

**Review by kevin3656 — APPROVED** [review]
Looks really good so awesome. Only thing I can say is maybe looking into Pressable vs TouchableOpacity

**Review by stephenhungg — APPROVED** [review]

File: src/app/_layout.tsx [hunk 2]
```diff
@@ -1,24 +1,47 @@
...
-      </Stack.Navigator>
+      </Stack.Navigator> */}
+      <Stack>
+        <Stack.Screen
+          name="my-tasks/page"
+          options={{
+            headerShown: true,
+            headerTitle: '',
+            headerShadowVisible: false,
+            headerStyle: { backgroundColor: '#fff' },
```
Comment (rachaelch3n) [inline src/app/_layout.tsx]: can we get rid of these lines and change headerShown to false? (remove headerTitle, headerShadowVisibe, and headerStyle)

File: src/app/_layout.tsx [hunk 3]
```diff
@@ -1,24 +1,47 @@
...
+          }}
+        />
+        <Stack.Screen
+          name="donation-details/[id]"
+          options={{
+            headerShown: true,
+            headerTitle: 'Donation Details',
+            headerStyle: { backgroundColor: '#f8f8f8' },
```
Comment (rachaelch3n) [inline src/app/_layout.tsx]: also remove these lines and change headerShown to false

File: src/app/donation-details/[id].tsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { Text, View } from 'react-native';
+import { useLocalSearchParams } from 'expo-router';
+
+export default function DonationDetails() {
+  const { id } = useLocalSearchParams<{ id: string }>();
+  return (
+    <View style={{ flex: 1, padding: 16 }}>
```
Comment (rachaelch3n) [inline src/app/donation-details/[id].tsx]: can we avoid inline styling and make a stylesheet for this instead?

`const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 16,
  },
});`

File: src/components/Button/Button.tsx
Comment (rachaelch3n) [inline src/components/Button/Button.tsx:1]: is there a reason you commented this out?

File: src/app/donation-details/_layout.tsx (hunk 1 above)
Comment (rachaelch3n) [reply src/app/donation-details/_layout.tsx]: yes can we remove it?

**Review by rachaelch3n — CHANGES_REQUESTED** [review]
yay overall really nice job with handling the logic for displaying the pages for each task! just a few minor tweaks and this PR should be good to go :)

File: src/app/my-tasks/index.tsx
Comment (rachaelch3n) [inline src/app/my-tasks/index.tsx:1]: one more thing and this is kinda on me but im gonna copy and paste what kyle said on stephen's pr regarding a change needed for the stack navigation:

"i think you almost have the right idea for expo router stack navigation, but some tweaks need to be made:

add a _layout.tsx file to each of the new screen folders (/login and /signup [/my-tasks for you]), and rename page.tsx to index.tsx
in the _layout.tsx file, you can past this as the content:
import { Stack } from 'expo-router';

function StackLayout() {
  return (
    <Stack>
      <Stack.Screen name="index" options={{ headerShown: false }} />
    </Stack>
  );
}

export default StackLayout;
this basically exports the index.tsx file as the screen itself, so we can go back to app/_layout.tsx and rename the Stacks to name="auth/login" and name="auth/signup" respectively.
3. this allows us to use router to route to auth/signup or auth/login **[/my-tasks]** instead of adding the /page at the end, which is kinda bad practice

lmk if this doesn't make sense, if you need more examples take a look at [Girls Write Now's codebase](https://github.com/calblueprint/girls-write-now)"

File: src/app/_layout.tsx [hunk 5]
```diff
@@ -1,24 +1,47 @@
...
   return (
     <SafeAreaProvider>
-      <Stack.Navigator screenOptions={{ headerShown: false }}>
+      {/* <Stack.Navigator screenOptions={{ headerShown: false }}>
         <Stack.Screen name="Index" component={IndexScreen} />
-      </Stack.Navigator>
+      </Stack.Navigator> */}
+      <Stack>
+        <Stack.Screen
+          name="my-tasks/page"
```
Comment (rachaelch3n) [inline src/app/_layout.tsx]: marking this line for when you address the layout structure. we can get rid of /page once you do this

**Review by rachaelch3n — CHANGES_REQUESTED** [review]

File: src/app/donation-details/_layout.tsx
Comment (rachaelch3n) [inline src/app/donation-details/_layout.tsx:1]: We need to move this entire component out of _layout.tsx and into [index].tsx, and keep _layout.tsx only for stack layout (navigation setup). Right now, this fie is doing two jobs -- setting up the navigation structure AND rendering the screen's actual UI, which should be reserved for [id].tsx. _layout.tsx is only intended to define navigation wrappers and shared layout, not screen content. Essentially, the only things we should define here is: `import { Stack } from 'expo-router';

export default function DonationLayout() {
  return (
    <Stack
      screenOptions={{
        headerShown: true,
        headerTitleAlign: 'center',
        headerTintColor: 'black',
      }}
    >
      <Stack.Screen
        name="index"
        options={{
          headerTitle: 'Donation Details',
          headerBackButtonDisplayMode: 'minimal',
        }}
      />
    </Stack>
  );
}
`
and move the rest to [id].tsx

You can reference [kevin's PR](https://github.com/calblueprint/replate/pull/4#pullrequestreview-3402554263) for help as he also had to do dynamic routing (although with different logic -- but definitely look for file structure).

**Review by rachaelch3n — CHANGES_REQUESTED** [review]

File: src/components/Button/Button.tsx
Comment (jeannineruiz) [reply src/components/Button/Button.tsx:1]: yeh it was giving me the error "Cannot find module '@rneui/themed' or its corresponding type declarations.ts(2307)" i did not touch it beforehand so i just commented it out for my sanity
